platform/files.go

```diff
@@ -279,38 +283,79 @@ type RunImageMetadata struct {
 	Mirrors []string `toml:"mirrors" json:"mirrors,omitempty"`
 }
 
-func (rm *RunImageMetadata) BestRunImageMirror(registry string) (string, error) {
+//go:generate mockgen -package testmock -destination ../testmock/access_checker.go github.com/buildpacks/lifecycle/platform AccessChecker
+type AccessChecker interface {
+	CheckReadAccess(repo string, keychain authn.Keychain) (bool, error)
```

Comment:
Could we maybe extend `[ImageHandler](https://github.com/buildpacks/lifecycle/blob/b6af8b55150542b20f896bfff8ed8fd47a0e1bc5/image/handler.go)` to include this? This was added recently, if you pull in the latest main you should see it (apologies in advance for the merge conflict)

Reply:
We tried, but the problem seems to be that creating an `ImageHandler` needs a `KeyChain` and the `Keychain` needs input from runMetadata which is not available before `resolve_inputs.go`.

Reply:
I see... that is a bummer. The issue that I could see with where `CheckReadAccess` is called is that we'd do this check even when exporting to a daemon or layout path, and those targets will fail. So we'll want the `AccessChecker` interface to have different implementations for those targets with basically a no-op `CheckReadAccess`.

Reply:
Thinking more about the comment above, if we are going to check read access to images in the daemon (for local builds) then resolving the inputs must be done after the client is constructed (in `cmd.Privileges()`). This unfortunately requires the re-wiring of a bunch of stuff that I tried to illustrate in this [draft PR](https://github.com/buildpacks/lifecycle/pull/1031/files). Unfortunately I couldn't think of a better way to get the access checker than just passing it to every `LifecycleInputsOperation`. If you can think of a better way to do it, I am all for it.

Reply:
> The issue that I could see with where `CheckReadAccess` is called is that we'd do this check even when exporting to a daemon or layout path, and those targets will fail.

I have probably missed the point, but we are using `CheckReadAccess` "only" on the `run-image` and the `run-image-mirrors`. So shouldn't the export target have nothing to do with that?

Reply:
@natalieparellano The more I thought of this and tried things out, the more it felt that `AccessChecker` and `image.Handler` should not be the same thing. The `AccessHandler` is actually only used for run-images (always remote, right?)

I tried something different. Calling it `ImageStrategy` (better names are more than welcome!) since that what it basically is imho. We want just another way to verify readability of images in the unit tests.

I also got rid of the according `gomock` mocks and used a simple test implementation. Which is a huge gain in regards to readability of the unit tests.

WDYT?

Please have a look, I pushed it a a separate commit, so it would be easy to throw it away again.